templates/pxelinux.cfg.j2

```diff
@@ -0,0 +1,65 @@
+# {{ ansible_managed }}
+
+path bin
+
+DEFAULT {% if menu.value.vesamenu|default(tftp_vesamenu) -%} vesa {%- endif -%} menu.c32
+PROMPT {{ menu.value.always_show_boot_prompt|default(tftp_always_show_boot_prompt) | int }}
+TIMEOUT {{ menu.value.timeout|default(tftp_timeout)*10 | round | int }}
+
+{% if menu.value.kbmap is defined or tftp_kbmap is defined -%}
+KBDMAP {{ menu.value.kbmap|default(tftp_kbmap) }}
+{%- endif %}
+
+{% if menu.value.title is defined -%}
+MENU TITLE {{ menu.value.title }}
+{%- endif %}
+{% if menu.value.tabmsg is defined -%}
+MENU TABMSG {{ menu.value.tabmsg }}
+{%- endif %}
+{% if menu.value.extra_entries is defined %}
+
+
+{% for entry in menu.value.extra_entries %}
+{{ entry }}
+{% endfor %}
+{% endif %}
+
+{% set indent_level = [0] %}
+{% for entry in menu.value.entries recursive %}
+
+{% if entry is string and entry == "separator" -%}
+MENU SEPARATOR
+
+{% elif entry.intmenu is defined %}
+{% if entry.label is defined %}
+LABEL {{ loop.depth }}_{{ loop.index }}
+	MENU LABEL {{ entry.label }}
+	MENU INDENT {{ indent_level[-1]*4 }}
+	MENU DISABLE
+{% endif %}
+{% if indent_level.append(indent_level.pop() + 1) %}{% endif %}{# workaround to maintain actual indent_level counter#}
```

Comment:
Why the if, is an execution like in line 27 not possible?

Reply:
I need to manipulate the indent_level variable, which is not possible by the set command in jinja2, since this generates always a new variable and the new value is lost at the next iteration.